src/main/java/io/supertokens/storage/mysql/queries/BulkImportQueries.java

```diff
@@ -0,0 +1,303 @@
+                                .generateCommaSeperatedQuestionMarks(bulkImportUsers.size()) + ")";
+
+                update(sqlCon, updateQuery, pst -> {
+                    int index = 1;
+                    pst.setString(index++, BULK_IMPORT_USER_STATUS.PROCESSING.toString());
+                    pst.setLong(index++, System.currentTimeMillis());
+                    pst.setString(index++, appIdentifier.getAppId());
+                    for (BulkImportUser user : bulkImportUsers) {
+                        pst.setObject(index++, user.id);
+                    }
+                });
+                return bulkImportUsers;
+            } catch (SQLException throwables) {
+                throw new StorageTransactionLogicException(throwables);
+            }
+        });
+    }
+
+    public static List<BulkImportUser> getBulkImportUsers(Start start, AppIdentifier appIdentifier,
+            @Nonnull Integer limit, @Nullable BULK_IMPORT_USER_STATUS status,
+            @Nullable String bulkImportUserId, @Nullable Long createdAt)
+            throws SQLException, StorageQueryException {
+
+        String baseQuery = "SELECT * FROM " + Config.getConfig(start).getBulkImportUsersTable();
+
+        StringBuilder queryBuilder = new StringBuilder(baseQuery);
+        List<Object> parameters = new ArrayList<>();
+
+        queryBuilder.append(" WHERE app_id = ?");
+        parameters.add(appIdentifier.getAppId());
+
+        if (status != null) {
+            queryBuilder.append(" AND status = ?");
+            parameters.add(status.toString());
+        }
+
+        if (bulkImportUserId != null && createdAt != null) {
+            queryBuilder
+                    .append(" AND (created_at < ? OR (created_at = ? AND id <= ?))");
+            parameters.add(createdAt);
+            parameters.add(createdAt);
+            parameters.add(bulkImportUserId);
+        }
+
+        queryBuilder.append(" ORDER BY created_at DESC, id DESC LIMIT ?");
+        parameters.add(limit);
+
+        String query = queryBuilder.toString();
+
+        return execute(start, query, pst -> {
+            for (int i = 0; i < parameters.size(); i++) {
+                pst.setObject(i + 1, parameters.get(i));
+            }
+        }, result -> {
+            List<BulkImportUser> bulkImportUsers = new ArrayList<>();
+            while (result.next()) {
+                bulkImportUsers.add(BulkImportUserRowMapper.getInstance().mapOrThrow(result));
+            }
+            return bulkImportUsers;
+        });
+    }
+
+    public static List<String> deleteBulkImportUsers(Start start, AppIdentifier appIdentifier,
+            @Nonnull String[] bulkImportUserIds) throws SQLException, StorageQueryException {
+        if (bulkImportUserIds.length == 0) {
+            return new ArrayList<>();
+        }
+
+        String selectQuery = "SELECT id FROM " + Config.getConfig(start).getBulkImportUsersTable()
```

Comment:
add a comment explaining why you are doing select first

Reply:
done